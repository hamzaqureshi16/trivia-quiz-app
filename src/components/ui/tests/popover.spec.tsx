import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Popover, PopoverTrigger, PopoverContent } from '../popover';

describe('Popover Component', () => {
  it('renders the Popover trigger', () => {
    render(
      <Popover>
        <PopoverTrigger>Open Popover</PopoverTrigger>
        <PopoverContent>Popover Content</PopoverContent>
      </Popover>,
    );
    expect(screen.getByText('Open Popover')).toBeInTheDocument();
  });

  it('does not show the Popover content by default', () => {
    render(
      <Popover>
        <PopoverTrigger>Open Popover</PopoverTrigger>
        <PopoverContent>Popover Content</PopoverContent>
      </Popover>,
    );
    expect(screen.queryByText('Popover Content')).not.toBeInTheDocument();
  });

  it('shows the Popover content when the trigger is clicked', () => {
    render(
      <Popover>
        <PopoverTrigger>Open Popover</PopoverTrigger>
        <PopoverContent>Popover Content</PopoverContent>
      </Popover>,
    );
    const trigger = screen.getByText('Open Popover');
    fireEvent.click(trigger);
    expect(screen.getByText('Popover Content')).toBeInTheDocument();
  });

  it('hides the Popover content when the trigger is clicked again', () => {
    render(
      <Popover>
        <PopoverTrigger>Open Popover</PopoverTrigger>
        <PopoverContent>Popover Content</PopoverContent>
      </Popover>,
    );
    const trigger = screen.getByText('Open Popover');
    fireEvent.click(trigger); // Open the popover
    fireEvent.click(trigger); // Close the popover
    expect(screen.queryByText('Popover Content')).not.toBeInTheDocument();
  });
});
