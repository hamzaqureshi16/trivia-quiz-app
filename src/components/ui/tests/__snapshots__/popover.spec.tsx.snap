// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover Component matches the snapshot 1`] = `
<button
  aria-controls="radix-«r5»"
  aria-expanded="false"
  aria-haspopup="dialog"
  data-slot="popover-trigger"
  data-state="closed"
  type="button"
>
  Open Popover
</button>
`;
